### ISSUE TITLE
HTTP POST is selected but GET is passed in request

### ISSUE THREAD

**Issue by syedhasanabbaszaidi** [a85/POSTMan-Chrome-Extension#8689]
**Describe the bug**
While making a POST request, POST is selected in drop down, the request goes with a GET request.
Version 7.26.0

**To Reproduce**
I have number of API calls , all of them are POST calls, I keep navigating between them and making calls. Once I am on my last call in the list request are going with GET while POST is selected in drop down.

Restarted the postman application and it worked fine.... Moved below to other API calls and again same issue.. I had to restart the app after some time.

**Expected behavior**
A POST must go as POST not GET call.

**Screenshots**
![image](https://user-images.githubusercontent.com/13048828/85226822-24910800-b3f7-11ea-9fda-3b3117f021bc.png)
 

**App information (please complete the following information):**
 Postman 7.26.0
Chrome Version 83.0.4103.106

**Additional context**

**Comment by dmytro-me** [comment]
Same problem. Postman Console displays that my POST is actually a GET request.

**Comment by DannyDainton** [comment]
Hey @syedhasanabbaszaidi

Could you confirm that you're still seeing the same behaviour when all the request tabs are saved (without the orange dots) please?

Have you used the duplicate tab option at all when creating the different requests?

Would you also be able to include an image of what you're seeing in the console for the request that's failing, please? 

Just trying to gather more information at this stage.

**Comment by dmytro-me** [comment]
May I respond instead? 

<img width="1259" alt="Screenshot 2020-06-21 at 19 05 02" src="https://user-images.githubusercontent.com/2362207/85267396-82fccb80-b475-11ea-8dd1-18a773f4681d.png">

Despite User-Agent string says, I'm on Postman Version 7.26.1 (7.26.1). 

I can duplicate an issue with a single unmodified request opened in Postman.

I noticed that in my case if I target localhost services, Postman sends POST request, but if I switch to remote (via Environment option) it sends GET.

Hope this helps.

**Comment by dmytro-me** [comment]
Just one more thing: my localhost service is on http, but remote accessible via http and https.  You may notice on screenshot that first POST tries http and then GET https:// . In my request I had 

```POST {{stat_host}}/URI```

In Environment I had stat_host specified without protocol (expect http).  When I specify protocol (https://myapi....com) everything works fine and POST is POST.  If I remove protocol, Postman sends GET instead of POST
Hope this helps.

**Comment by syedhasanabbaszaidi** [comment]
Yes, i can also confirm that it has to  do to with https.
Initiallly I was working on http localhost, later I changed to my server but with only http not https. On console I can see https request but with GET . once I made my call to https in URL section correct POST was sent out.

![image](https://user-images.githubusercontent.com/13048828/85277492-20212980-b4a1-11ea-8f61-df5de48c91ac.png)

**Comment by appurva21** [comment]
Hey guys!! 
I see your initial requests are getting redirected with 301 status code and according to the RFC [here](https://tools.ietf.org/html/rfc7231#section-6.4.2), it is the expected behavior. 

> For historical reasons, a user agent MAY change the request
      method from POST to GET for the subsequent request.  If this
      behavior is undesired, the 307 (Temporary Redirect) status code
      can be used instead.

So, to preserve the HTTP method on redirect, the ideal status code should be 307 or 308.

For your use case, you can go to the  Settings tab inside the request and enable the "Follow original HTTP Method" setting for that request.

![image](https://user-images.githubusercontent.com/24985700/85304603-90dc3c00-b4c9-11ea-99b1-1585dea790fc.png)

**Comment by dmytro-me** [comment]
Nice finding! However, in case of Postman, the API tester, I suppose this is wrong behaviour, especially since sent GET request looks malformed for me. While body of the request is kept, the Content-Type and Content-Length headers are missing.

Probably for historical reason you need to keep this, but at least some kind warning should be provided.

**Comment by syedhasanabbaszaidi** [comment]
Absolutely, from the postman's interface, it looks like it is a POST (as selected in the dropdown) but console shows a GET. Either a warning or change to the dropdown method has to come.

**Comment by vvs11** [comment]
@dmytro-me @syedhasanabbaszaidi I agree with you that it makes sense to notify to the user somehow that the change has happened. I am marking it as a feature request for our tracking.

**Comment by brenoca** [comment]
@syedhasanabbaszaidi
you are missing a slash at the end of the URL, you must add '/' after '../addProfile'
`http://kwantu.support/api/addProfile/`
I always forget this as well :/ (very annoying)

![image](https://user-images.githubusercontent.com/72074654/130286667-0b7dd55d-8008-4a7e-af8c-a7626c263c29.png)

**Comment by codysechelski** [comment]
This is happening to me as well, but here's the kicker. It's only happening in one Workspace...

I have a collection that lives in a workspace I made for my last company. Now at my new company, I wanted to move that collection into a new workspace.

There is probably a better way to do this, but here's how I did it (just in case any of this is important). Form Workspace A, I exported my globals, environments, and the collection. Then I switched to Workspace B and imported all three files. Everything seems to have worked just fine.

However, I noticed that even though one of the requests in that collection is a POST, it is being sent as a GET.

So I went back to Workspace A, and it is working just fine. It sends a POST as it should. I checked a few more requests from this collection that are also something other than a GET, and it is consistent. All of them from Workspace A are functioning as expected and everything from Workspace B is sending every request as a GET, regardless of the verb that I select in the dropdown.

I double, triple, and quadruple-checked that the endpoints are identical, the headers and params are identical, the body is identical,  the globals and environment variables are identical (both keys and values).

I also tried just creating a brand new request in Workspace A that uses the same verb, endpoint, headers, params, and so forth and it worked just fine. I manually created the exact same thing in Workspace B and it sent a GET instead of a POST, which was the verb that I selected.

**Comment by shubhbhargav** [comment]
@codysechelski It might be a different issue that you are facing. Can you please open a ticket with us by sending an email to `help@postman.com` with some details around the workspace IDs as well as maybe a sample collection to reproduce the issue?

**Comment by erlangparasu** [comment]
facing same problem on Windows

**Comment by erlangparasu** [comment]
i tried delete POST option, then manually type `POST` on the input area. But `--request POST` still does not appear. if i change to `POSTABC` , `--request POSTABC` appear (cURL generated)

**Comment by dhebel** [comment]
Maybe too late but... to anyone reading this with he same problem: Try with HTTPS instead of HTTP.
POST method usually requires HTTPS instead of HTTP when consumed from a public route outside of your IP/localhost.

**Comment by Faeeth** [comment]
try to use "BODY" tab instead of "PARAMS". It works for me

**Comment by pmclachlan** [comment]
I had to "Follow original HTTP Method" in the settings for the request.  Huge waste of time trying to find this option, in my opinion this is incredibly unfriendly default behavior for a developer tool where, when you have a problem (that your POST's are simply not working at all), your default mindset is likely to assume it's your own code's fault.
